File: idtx_pos_receipt/models/pos_order_line.py
# -*- coding: utf-8 -*-
from odoo import api, fields, models

class PosOrderLine(models.Model):
    _inherit = 'pos.order.line'

    lot_color_name = fields.Char("Lot Color", compute='_compute_lot_color_name')

    @api.depends('pack_lot_ids.lot_name')
    def _compute_lot_color_name(self):
        for line in self:
            if line.pack_lot_ids:
                # Buscamos el lote por nombre
                lot = self.env['stock.lot'].search([
                    ('name', '=', line.pack_lot_ids[0].lot_name),
                    ('product_id', '=', line.product_id.id),
                ], limit=1)
                if lot.color_code and lot.color_name:
                    line.lot_color_name = ('[' + lot.color_code + '] ' + lot.color_name) or ''
                else:
                    line.lot_color_name = ''
            else:
                line.lot_color_name = ''

    @api.model
    def _load_pos_data_fields(self, config):
        return super()._load_pos_data_fields(config) + ['lot_color_name']

File: idtx_pos_lot_color/static/src/js/pos_order_line.js
/** @odoo-module **/
import { PosOrderline } from "@point_of_sale/app/models/pos_order_line";
import { patch } from "@web/core/utils/patch";

patch(PosOrderline.prototype, {
    get lotColorName() {
        if (!this.pack_lot_ids?.length) return "";
        const lotName = this.pack_lot_ids[0].lot_name;
        if (!lotName || !this.models) return "";
        const lot = (this.models["stock.lot"] || []).find(l =>
            l.name === lotName
        );
        if (!lot) return "";
        return (lot?.color_code && lot?.color_name)
            ? `[${lot.color_code}] ${lot.color_name}`
            : "";
    },
});